### DISCUSSION TITLE
GitHub should have an option to disable the suggestion of "Compare & Pull Request"

### DISCUSSION THREAD

**Discussion by SCodify** [community/community#116576] [Pull Requests]
### Select Topic Area

General

### Body

Supposedly this is a GitHub feature to make it easier to create Pull Requests after pushing to a branch. Currently, GitHub does not provide an option to disable these suggestions in the user interface. These suggestions are shown to all users who have write permissions on the repository when they push to a branch.
While one cannot disable these messages, one can choose to ignore them if one does not wish to create a Pull Request at that time. I think that being able to disable these suggestions should be an option because sometimes it is annoying and often accumulate messages from "Compare & Pull Request".

**Comment by SCodify** [comment]
Another disadvantage is that if you touch that button you want to merge directly to the main branch.

**Comment by SCodify** [comment]
![Captura de pantalla 2024-04-02 164938](https://github.com/community/community/assets/104208569/25649b32-e813-4d78-9880-b44debc8fce5)
This does not generate a good user experience.

**Comment by queenofcorgis** [comment]
Thanks for posting in the GitHub Community, @SCodify ! We're happy you're here. You are more likely to get a useful response if you are posting your question in the applicable category, the Discussions category is solely related to conversations around the GitHub product Discussions. I've gone ahead and moved it for you. Good luck!

**Comment by meghasharma123** [comment]
still not resolved. Please check, its not a good user experience.

**Comment by KuryKat** [comment]
I completely agree with this suggestion. Not all repositories use branches as staging areas for PRs. Many projects use branches for version management... and yes, I know tags exist for versioning, but branches work better for certain workflows, especially when:

- You're maintaining multiple active versions simultaneously
- Different branches need their own tags
- You're supporting different versions of dependencies across branches

In these cases, the main branch might just hold documentation, while version-specific code lives on separate branches that get linked in releases. For these workflows, the "Compare & Pull Request" suggestion isn't just unnecessary... it's actively annoying since you're never planning to merge those branches. Having an option to disable it would be a huge UX improvement.

**Comment by JoaquinGasparMedinaTiznado** [comment]
I have the same suggestion. It's 2026 and apparently there's still no solution.

**Comment by quangnh89** [comment]
I have a similar suggestion too. Currently, my solution is to create a pull-request and close it immediately. Everything works fine again.

**Comment by hannesjohansson** [comment]
Would love to disable this for certain branches as well (or completely)

**Comment by jmason888** [comment]
Also for orphan branches that aren't called "gh-pages"...
